form.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+    function func(String $text, int $size, &$valid, &$errors){
+        $tmp = $_POST["$text"];
+        if(!$tmp){
+            $errors["$text"] = "Полето е задължително."; 
+        } elseif(strlen($tmp) > $size ){
+            $errors["$text"] = "Полето има максимална дължина $size символа.";
+        } else {
+            $valid[$text] = $tmp;
+        }        
+    }
+
+    $valid = array();
+    $errors = array();
+    
+    if( $_POST ){
+        func("title", 150, $valid, $errors);
+        func("teacher",200, $valid, $errors);
+
+        $description = $_POST['description'];
+        if(!$description){
+            $errors['description'] = "Описанието е задължително."; 
+        } elseif(strlen($description) < 10 ){
+            $errors['description'] = "Описанието има минимална дължина 10 символа.";
+        } else {
+            $valid['description'] = $description;
+        }
+
+        $group = $_POST['group'];
+        if(!$group){
+            $errors['group'] = "Не сте избрали група.";
+        } else {
+            $valid['group'] = $group;
+        }
+
+        $credits = $_POST['credits'];
+        if (!$credits){
+            $errors['credits'] = "Не сре задали кредити.";
+        } elseif($credits < 0 || $credits > 20){
+            $errors['credits'] = "Кредити могат да са от 0 до 20.";
+        } else {
+            $valid['credits'] = $credits;
+        }
+
+        if(count($valid) == 5 ){
+
+            $filename = 'results.txt';
+            file_put_contents($filename, "Име на предмета: ", FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, $_POST['title'], FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, "\n", FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, "Преподавател: ", FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, $_POST['teacher'], FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, "\n", FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, "Описание: ", FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, $description, FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, "\n", FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, "Група: ", FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, $group , FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, "\n", FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, "Kредити: ", FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, $credits, FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, "\n", FILE_APPEND | LOCK_EX);
+            file_put_contents($filename, "\n", FILE_APPEND | LOCK_EX);
+        } 
+    }
```

Comment:
P.S.: When copy-pasting solutions try also to test them and think how you can improve them.